File: tools/src/test/java/de/medizininformatikinitiative/medgraph/tools/UnitTest.java
package de.medizininformatikinitiative.medgraph.tools;

import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.BeforeEach;
import org.mockito.MockitoAnnotations;

/**
 * @author Markus Budeus
 */
public class UnitTest {

	private AutoCloseable mocks;

	@BeforeEach
	void setUpMocks() {
		mocks = MockitoAnnotations.openMocks(this);
	}

	@AfterEach
	void closeMocks() throws Exception {
		mocks.close();
	}

}

File: composeApp/src/commonTest/kotlin/de/medizininformatikinitiative/medgraph/ui/UnitTest.kt
package de.medizininformatikinitiative.medgraph.ui

import org.junit.jupiter.api.AfterEach
import org.junit.jupiter.api.BeforeEach
import org.mockito.MockitoAnnotations

/**
 * Superclass for unit tests, takes care of mockito annotations.
 *
 * @author Markus Budeus
 */
open class UnitTest {

    private lateinit var mocksCloseable: AutoCloseable

    @BeforeEach
    fun setupMocks() {
        mocksCloseable = MockitoAnnotations.openMocks(this)
    }

    @AfterEach
    fun tearDownMocks() {
        mocksCloseable.close()
    }

}